vip-helpers/vip-elasticsearch.php

```diff
@@ -62,18 +62,36 @@ function es_api_search_index( $args ) {
 
     $args['blog_id'] = absint( $args['blog_id'] );
 
-    $service_url = 'https://public-api.wordpress.com/rest/v1/sites/' . $args['blog_id'] . '/search';
+    $endpoint = sprintf( '/sites/%s/search', $args['blog_id'] );
+    $service_url = 'https://public-api.wordpress.com/rest/v1' . $endpoint;
+
+    $do_authenticated_request = false;
+    if ( class_exists( 'Jetpack_Client' )
+            && isset( $args['authenticated_request'] )
```

Comment:
Maybe worthwhile to add a new filter for args (to allow setting the `authenticated_request` arg) and a filter for the results (and stripping out posts that don't match user permissions) that can be easily added/removed.

The use case would be when using something like `es-wp-query` (or other contexts) where we may not have the ability to manipulate the args passed into or returned from `es_api_search_index`.